### ISSUE TITLE
Add Email Wrapper shared block

### ISSUE THREAD

**Issue by joshuatf** [Aljullu/woocommerce-gutenberg-emails#9]
The email header should be a customizable shared block.  This is called via an action hook:

`do_action( 'woocommerce_email_header', $email_heading, $email );`

**Comment by Aljullu** [comment]
Looking at this and #10, I'm not sure the header and footer should be stand-alone blocks. [`woocommerce_email_header`](https://github.com/woocommerce/woocommerce/blob/master/templates/emails/email-header.php) is used to open several HTML tags (`<html>`, `<body>`, `<table>`, etc.) and it leaves them open. They are later closed in [`woocommerce_email_footer`](https://github.com/woocommerce/woocommerce/blob/master/templates/emails/email-footer.php). From what I understand, Gutenberg blocks should be in some way self-contained (aka they should close every HTML tag they open), is that right?

In that case, I think there are two solutions here:
- Create a block named _Email wrapper_ or something like this, similar to the _Columns_ block or other blocks that can contain child blocks. Then, all other email blocks must be added inside the wrapper. This _Email wrapper_ block would have the markup from the header and footer.
- Attach the header/footer HTML when the block is rendered. I guess this option would the easiest one but would miss the point of seeing the complete email template when editing it in Gutenberg.

What do you think?

**Comment by ryelle** [comment]
What about a combination of both– a parent block "Email Wrapper" to show the header in the edit mode, but it wouldn't include the `html` or `head` tags, it would maybe start from [the div on line 30](https://github.com/woocommerce/woocommerce/blob/master/templates/emails/email-header.php#L30). Saving the email would save everything in that div. Then when we're ready to build the email to send it, we wrap it with the other tags.

I'm not sure Gutenberg would even let us use html/head/body, anyway.

**Comment by joshuatf** [comment]
Good point @Aljullu.  I think @ryelle's solution is a good hybrid and we can append invalid HTML elements in the editor (like `html` and `head`) during email processing.

**Comment by Aljullu** [comment]
That sounds like a good solution @ryelle. I will close #10 and rename this issue accordingly.
